### PR TITLE
async support check mode

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Feature Pull Request

### CODE REVIEW

**Review by Akasurde — APPROVED** [review]

**Review by jborean93 — APPROVED** [review]
I'll have to update `async_status.ps1` in `ansible.windows` but will hold off until this is merged so I can actually test it there.

**Pull request merged**
